### ISSUE TITLE
LibreSignal will die in 9 days unless it is updated.

### ISSUE THREAD

**Issue by herbsmn** [LibreSignal/LibreSignal#49]
Is this the end of this project? Anyone want to get the band back together for one last show?

**Comment by tuxayo** [comment]
> What to do after LibreSignal was abandoned?

https://github.com/LibreSignal/LibreSignal/wiki/What-to-do-after-LibreSignal-was-abandoned%3F

It's the end of the app itself but not the end of the project. The related work is in the process of being upstreamed, there is even a bounty on the task!

https://www.bountysource.com/issues/35722527-create-proper-pull-request-to-add-libresignal-s-websocket-support-to-ows-signal

The actual ticket/issue is ~~#37~~ (whoops, wrong issue) → https://github.com/LibreSignal/LibreSignal/issues/43

Don't expect this to happen soon but it's definitely moving forward.

**Comment by mimi89999** [comment]
Maybe I will update LS this week if i have time, for those who still can't move to Conversations or something else.

**Comment by h-2** [comment]
@mimi89999 That would be great. If we don't get upstream within the next months, I will stop using it, but it seems like it could actually happen in the next months...

**Comment by Lyle-Tafoya** [comment]
So I won't be able to continue encrypted communications with my girlfriend who uses vanilla Signal Messenger unless I convince her to use a different app now (or unless I install google on my phone)? Please confirm whether I am understanding this correctly.

**Comment by herbsmn** [comment]
that'd be greatly appreciated @mimi89999!

**Comment by mimi89999** [comment]
Oh no, they added more GMS lib stuff:

https://github.com/WhisperSystems/Signal-Android/commit/94a29e375f1209e39f58b7f52c1ec1ccb5ae75ce

`com.google.android.gms.security.ProviderInstaller`

**Comment by mimi89999** [comment]
BTW, wasn't there a comment by @Lyle-Tafoya ?

**Comment by Lyle-Tafoya** [comment]
Yeah, I commented before reading the entire link you posted. I thought there would be no way to continue encrypted communications with my girlfriend whom I convinced to use Signal awhile back (but only managed to do so because of the seamless SMS integration).
I see there is a way for me to run Signal with microG GmsCore, but I have not yet succeeded at getting this working on my phone since my phone is not supported by Xposed or Needle (looking into Haystack right now, but I haven't managed to successfully follow the instructions yet). I doubt I will be able to find the time to get that working before this version of LibreSignal expires.

Also, depending on how the alternative options look, I may be willing to pay for hosting costs so we can fork LibreSignal and move it away from Signal's servers (I'm not sure how expensive this is. If it's too expensive, I might be willing to pool together funds with someone else). If you have any guide on how I can build this project (preferably without any Windows or OS X specific instructions), I can start playing around with it in my spare time. However, that wouldn't be something that I would likely be able to do in a short period of time.

tl;dr I would greatly appreciate at least 1 more update and I might be willing to continue this project in my spare time if it will otherwise die.

**Comment by herbsmn** [comment]
> Oh no, they added more GMS lib stuff:
> WhisperSystems@94a29e3
> com.google.android.gms.security.ProviderInstaller

Is this new GMS lib stuff likely going to delay an updated LS? There's now less than 2 days before the app stops working.

**Comment by h-2** [comment]
> Is this new GMS lib stuff likely going to delay an updated LS? There's now less than 2 days before the app stops working.

One could maybe just take a snapshot from before the change, seeing that this is not going to be a permanent solution anyway.

**Comment by herbsmn** [comment]
can you explain the snapshot idea a little bit more @h-2?

**Comment by mimi89999** [comment]
Update only to the version before that change/feature was introduced.

**Comment by mimi89999** [comment]
I think it is really the time to move to Conversations or to something else...

**Comment by herbsmn** [comment]
> Update only to the version before that change/feature was introduced.

I don't understand what this means.

**Comment by h-2** [comment]
> I think it is really the time to move to Conversations or to something else...

@mimi89999  I think Conversations and Signal have totally different usecases. I use and like conversations, but for most of my contacts it is absolutely not an option and Signal is.
Since moxie seems to be open to a PR and the webrtc branch would also solve all the issues with redphone, it seems like we are actually making progress here.

If I need to disconnect from signal network now, it will be really difficult and confusing to explain and then maybe go back in some time. So if we get another 3month extension it would be realllyyy great!! :heart: :pray: 

Thanks!

> I don't understand what this means.

@herbsmn Maybe the details are not so important. It would work without the "feature", at least for the immediate future.

**Comment by mimi89999** [comment]
@h-2 You could also buy yourself time by removing the time bomb from LS.

**Comment by tuxayo** [comment]
> Oh no, they added more GMS lib stuff:
> https://github.com/WhisperSystems/Signal-Android/commit/94a29e375f1209e39f58b7f52c1ec1ccb5ae75ce
> com.google.android.gms.security.ProviderInstaller

That might be off topic but does anyone know what this new stuff is used for? After a quick search I'm still confused about what this ProviderInstaller is for.

**Comment by mimi89999** [comment]
@tuxayo For fighting censorship of Signal.

**Comment by herbsmn** [comment]
@mimi89999 I am an activist, not a programmer, but I'm willing to try to make a version that removes the time bomb. Can you provide me with a couple hints on how this can be done? Just so I understand, is the current LibreSignal version updated to the Signal version right before ProviderInstaller was introduced? If not, is it just as easy to update LS as it is to remove the time bomb from the current version?

**Comment by mimi89999** [comment]
https://github.com/LibreSignal/LibreSignal/blob/master/src/org/thoughtcrime/securesms/util/Util.java#L340

It is set to 90 days from the build date.

**Comment by h-2** [comment]
> It is set to 90 days from the build date.

which is why even a "simple" rebuild instead of an actual update would workaround the timestamp problem for people right now...

**Comment by herbsmn** [comment]
@mimi89999 i didn't realize that the time bomb resets itself after a simple rebuild. so one doesn't need to remove the time bomb if they don't plan on using LS for more than 90 days after the rebuild?

**Comment by herbsmn** [comment]
is this the way to build LibreSignal too? https://github.com/WhisperSystems/Signal-Android/blob/master/BUILDING.md

Looks a tad complicated. Any chance you could do one last build and extend the timebomb @mimi89999?

**Comment by hackoder** [comment]
Here is what I was able to do:

- Use the instructions here to check out the code: https://github.com/LibreSignal/LibreSignal/blob/master/BUILDING.md
- Since I was upgrading from the eutopia.cz experimental repository (which uses package name org.thoughtcrime.securesms instead of org.libresignal), I switched applicationId to 'org.thoughtcrime.securesms' from 'org.libresignal' in build.gradle.
- The 3 month limit seems to be set from the date of the last commit, so I added a test commit to bump it up to today.
- Build the apk: ./gradlew build
- Use instructions here to sign the apk: https://github.com/WhisperSystems/Signal-Android/blob/master/BUILDING.md

Couldn't install it directly because my signature was different from the eutopia.cz repository. So, created a backup using [oandbackup](https://f-droid.org/repository/browse/?fdid=dk.jens.backup). Uninstalled existing LibreSignal, installed my version, then restored backup. 

This seems to be have worked for me. I have uploaded my build here if anyone wants to give it a shot: https://github.com/hackoder/LibreSignal/releases

Note that because of different package names (org.libresignal vs org.thoughtcrime.securesms) and different signatures, the process will probably be a little complicated so have a backup. Ideally somoene should do a recompile and sign/upload to the eutopia.cz repository.

**Comment by mimi89999** [comment]
Only @xmikos can sign LibreSignal.

**Comment by mimi89999** [comment]
> The 3 month limit seems to be set from the date of the last commit

Oh, yes...
https://github.com/WhisperSystems/Signal-Android/blob/f53d64ab7a77d4d79723729bfc4fba9685703c61/build.gradle#L197

**Comment by h-2** [comment]
@xmikos pretty please?

**Comment by hackoder** [comment]
> Only @xmikos can sign LibreSignal.

Yes, that's why I had to backup/uninstall/reinstall/restore. Though there are other ways to skip signature verification..

**Comment by mimi89999** [comment]
@hackoder 
> Though there are other ways to skip signature verification..

Nooooo... Not the xposed module...

**Comment by hackoder** [comment]
> @mimi89999: Nooooo... Not the xposed module...

Haha yeah, exactly why I skipped it.

**Comment by Lyle-Tafoya** [comment]
I tried to get this project to build last night but encountered some build errors on the ./gradlew clean step. Anyone else having better luck? Maybe tonight I'll try doing it all through Android Studio, but I suspect it's probably a library version conflict with the Linux distro I use.

**Comment by mimi89999** [comment]
@Lyle-Tafoya 
> Maybe tonight I'll try doing it all through Android Studio

Use gradle

> I tried to get this project to build last night but encountered some build errors on the ./gradlew clean step. Anyone else having better luck?

If you don't show your log, there is little chance, that somebody will be able to help you.

**Comment by h-2** [comment]
switched to noise ....

**Comment by Lyle-Tafoya** [comment]
@mimi89999 I wasn't so much asking for help as much as I was posting a status update and looking to see where other people are at currently. I work full time so I won't be able to even gain access to the log until later tonight.

**Comment by herbsmn** [comment]
@hackoder, I backed up my old libresignal on Replicant 4.2 with oandbackup and then updated to Replicant 6.0 because I couldn't install your build on 4.2. I then restored the backup and am able to see all of my old messages. The trouble is, I can't seem to send any messages. any change you or @mimi89999 might know why? @xmikos if you happen to get a chance to put this on your experimental repo for ease of use, it would be appreciated, but no worries if you'd rather not.

**Comment by herbsmn** [comment]
oh, and thanks for publishing your build @hackoder!

**Comment by mimi89999** [comment]
I haven't changed anything. As always, logs are appreciated.

**Comment by herbsmn** [comment]
damn, after hours of trying to figure it out I finally get it sorted out right after messaging you guys about it.

**Comment by herbsmn** [comment]
I had to toggle off and on "LibreSignal messages and calls" in Advanced settings, which is fairly non-descript thing to know to toggle. This was because I registered the app with a different key before restoring the backup. Thanks to all of you guys for your work on LibreSignal.

**Comment by scytho** [comment]
I got microG with the Blankstore working, backed up LibreSignal, uninstalled it, installed Signal and restored the app data. That worked, but Signal won’t let me register until I “update the Google Play Services”.

**Comment by ale5000-git** [comment]
@scytho: Because you haven't choose the latest version of microG (currently 0.2.4-55)

**Comment by scytho** [comment]
Thanks, that did the trick. Except one snag: I’m using Signal Desktop as well. I had to re-pair Signal and Signal Desktop. Now, messages I write with Signal Desktop are being synchronised to my phone, but not the other way round.

**Comment by ale5000-git** [comment]
@scytho: I suggest to report [here](https://github.com/microg/android_packages_apps_GmsCore/issues) and post a logcat.

**Comment by scytho** [comment]
You think it’s a problem with microG? I suspected OWS Signal.

**Comment by ale5000-git** [comment]
@scytho: I'm not sure, but it is better to open a "separate" issue to keep track of the problem.
If you post a message between other 1000 messages noone will notice it.

**Comment by varac** [comment]
Hej, I also switched to plain Signal using [microG](https://microg.org/) (latest, 0.2.4-64).
So far it looks like it works quite well. Installation is quite challenging (including patching in [signature spoofing](https://github.com/microg/android_packages_apps_GmsCore/wiki/Signature-Spoofing) ) but once it's done it's fine.
Still looking forward for a pure websocket-implementation as long-term solution...

**Comment by ghost** [comment]
@herbsmn Perhaps it's time to close this issue, since LibreSignal did not die on Jan 31, 2017? I have started a new discussion about the future of LibreSignal now that Signal is Google-free: https://github.com/LibreSignal/LibreSignal/issues/52

**Comment by herbsmn** [comment]
Yeah, I was wrong. I should have said that the experimental repo was dying, not LibreSignal. Closed.
